cocos/core/memop/pool.ts

```diff
@@ -49,12 +51,18 @@ export class Pool<T> extends ScalableContainer {
      * @param dtor @en The finalizer of element, it's invoked when this Pool is destroyed or shrunk if
      * it is valid.
      * @zh 元素的析构器。如果存在的话，当对象池销毁或者缩容时，会使用该析构器。
+     * @param shrinkThreshold @en The container is shrink only if the size of the container exceeds the shrinkThreshold,
+     * and the size of the container after reduction is greater than or equal to the shrinkThreshold. The value equals elementsPerBatch
+     * if not value is passed.
+     * @zh 只有容器的数量大于 shrinkThreshold 时才缩容，缩减后的容器大小小于等于 shrinkThreshold。如果没有传入值，那么它的值和 elementsPerBatch 相同。
      */
-    constructor (ctor: () => T, elementsPerBatch: number, dtor?: (obj: T) => void) {
+    constructor (ctor: () => T, elementsPerBatch: number, dtor?: (obj: T) => void, shrinkThreshold?: number) {
         super();
         this._ctor = ctor;
         this._dtor = dtor || null;
         this._elementsPerBatch = Math.max(elementsPerBatch, 1);
+        this._shrinkThreshold = shrinkThreshold || this._elementsPerBatch;
+        this._shrinkThreshold = max(this._shrinkThreshold, 1);
```

Comment:
shrinkThreshold may be negative, so should not do it like this.

Reply:
It should be the same?  If it's negative, it's also true?

Reply:
Yep, it should be the same. Indeed I will merge them as  `this._shrinkThreshold = shrinkThreshold || max(this._elementsPerBatch, 1);' since Javascript usually use `||` instead. I wrote `? :` first, and VSCode change to use `||` when save the file.

Reply:
Oh, your modification is right.
Fixed.